src/TerminalFighter/MainCharacter/MainCharacter.cpp

```diff
@@ -45,8 +45,19 @@ void MainCharacter::draw(I_GraphicsHandler& graphics) {
     for (auto& weapon : weapons_) {
         weapon->draw(graphics);
     }
-
-    graphics.draw(texture_, (int)x_pos(), (int)y_pos(), GraphicPriority::MIDDLE, true);
+    SDL_Rect src_rect;
+    SDL_QueryTexture(texture_, NULL, NULL, &src_rect.w, &src_rect.h);
+    src_rect.x = 0;
+    src_rect.y = 0;
```

Comment:
This is pretty ugly for everything that needs to draw something IMO. This is worth adding an overloaded function for since Graphics Handler is capable of finding this information on its own.